From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: tr7zw <tr7zw@live.de>
Date: Tue, 7 Dec 2021 20:53:10 -0500
Subject: [PATCH] Global Eula file

Original code by Yatopia, licensed under MIT License

Updated to 1.18 by: Steven <steven@nevets.tech>

diff --git a/src/main/java/net/minecraft/server/Eula.java b/src/main/java/net/minecraft/server/Eula.java
index a1d5c0f8fe2adb2ee56f3217e089211ec7c61eb0..00008b0dc478cccd710b1d82f5158f02f6271f03 100644
--- a/src/main/java/net/minecraft/server/Eula.java
+++ b/src/main/java/net/minecraft/server/Eula.java
@@ -16,12 +16,26 @@ public class Eula {
 
     public Eula(Path eulaFile) {
         this.file = eulaFile;
-        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || this.readFile();
+        this.agreed = SharedConstants.IS_RUNNING_IN_IDE || globalEula() || this.readFile(eulaFile);
     }
 
-    private boolean readFile() {
+    // Sugarcane start - global eula file
+    private boolean globalEula() {
+        java.io.File globalEula = new java.io.File(System.getProperty("user.home"), "eula.txt");
+
+        if (globalEula.exists()) {
+            System.out.println("Loaded global eula file from " + globalEula.getAbsolutePath());
+            return readFile(globalEula.toPath());
+        } else {
+            System.out.println("No global eula file found at " + globalEula.getAbsolutePath());
+        }
+        return false;
+    }
+    // Sugarcane end
+
+    private boolean readFile(java.nio.file.Path path) {
         try {
-            InputStream inputStream = Files.newInputStream(this.file);
+            InputStream inputStream = Files.newInputStream(path);
 
             boolean var3;
             try {
